graftabledag/graftabledag.go

```diff
@@ -89,9 +89,12 @@ func (gd *GraftedDag) resolveRecursively(ctx context.Context, path chaintree.Pat
 
 	didPaths := make([]chaintree.Path, 0)
 	values := make([]interface{}, 0)
+	returnSingle := false
 
 	switch v := value.(type) {
 	case string:
+		returnSingle = true
```

Comment:
I think I just realized in reviewing this that it probably needs to apply to any basic single value: bools, ints, etc --- so maybe need to flip the flag to `returnMulti` and check for slices or maps?